### PR TITLE
✨ Add `description` parameter to all the security scheme classes, e.g. `APIKeyQuery(name="key", description="A very cool API key")`

### PR DESCRIPTION
**Pull request by hylkepostma**
My suggestion is to add a description parameter to the `__init__` methods of `APIKeyQuery(SecurityBase):`, `APIKeyHeader(SecurityBase):` and `APIKeyCookie(SecurityBase):` and use it to initialize the `APIKey` object. This allows for a description in the `securitySchemes` objects of the `openapi.json` and therefore in the Swagger and Redoc documentation.

`class SecurityBase(BaseModel):` already has an attribute `description: Optional[str] = None`.
`class APIKey(SecurityBase):` inherits from SecurityBase.

The classes `APIKeyQuery(SecurityBase):`, `APIKeyHeader(SecurityBase):` and `APIKeyCookie(SecurityBase):` initialize an `APIKey` object in their `__init__` method. They initialize it with some kwargs and the `name` parameter, but currently without a description parameter.

Example: using `APIKeyQuery(name="X-Auth-Key", auto_error=False, description="My description")` would look likt this in Redoc:
image w:400

### CODE REVIEW

File: fastapi/security/api_key.py [hunk 1]
```diff
@@ -12,8 +12,17 @@ class APIKeyBase(SecurityBase):
 
 
 class APIKeyQuery(APIKeyBase):
-    def __init__(self, *, name: str, scheme_name: str = None, auto_error: bool = True):
-        self.model: APIKey = APIKey(**{"in": APIKeyIn.query}, name=name)
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str = None,
```
Comment (MashhadiNima) [inline fastapi/security/api_key.py:19]: `description: Optional[str] = None,`

File: fastapi/security/api_key.py [hunk 2]
```diff
@@ -12,8 +12,17 @@ class APIKeyBase(SecurityBase):
...
+        scheme_name: str = None,
```
Comment (MashhadiNima) [inline fastapi/security/api_key.py:20]: `scheme_name: Optional[str] = None,`

File: fastapi/security/api_key.py [hunk 3]
```diff
@@ -30,8 +39,17 @@ async def __call__(self, request: Request) -> Optional[str]:
...
 
 class APIKeyHeader(APIKeyBase):
-    def __init__(self, *, name: str, scheme_name: str = None, auto_error: bool = True):
-        self.model: APIKey = APIKey(**{"in": APIKeyIn.header}, name=name)
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str = None,
+        scheme_name: str = None,
```
Comment (MashhadiNima) [inline fastapi/security/api_key.py:47]: `        description: Optional[str] = None,
        scheme_name: Optional[str] = None,
	
`

File: fastapi/security/api_key.py [hunk 4]
```diff
@@ -48,8 +66,17 @@ async def __call__(self, request: Request) -> Optional[str]:
...
 
 class APIKeyCookie(APIKeyBase):
-    def __init__(self, *, name: str, scheme_name: str = None, auto_error: bool = True):
-        self.model: APIKey = APIKey(**{"in": APIKeyIn.cookie}, name=name)
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str = None,
+        scheme_name: str = None,
```
Comment (MashhadiNima) [inline fastapi/security/api_key.py:74]: `   
description: Optional[str] = None,   
scheme_name: Optional[str] = None,
`

File: fastapi/security/api_key.py (hunk 2 above)
Comment (MashhadiNima) [inline fastapi/security/api_key.py:20]: `
        description: Optional[str] = None,
        scheme_name: Optional[str] = None,

`

File: fastapi/security/api_key.py (hunk 3 above)
Comment (hylkepostma) [reply fastapi/security/api_key.py]: @MashhadiNima, thanks for the review. I've applied your improvements.

Comment (MashhadiNima) [reply fastapi/security/api_key.py]: you're welcome. 👨🏻‍💻🍻

**Comment by tiangolo** [comment]
Thanks @hylkepostma! :coffee: 

And thanks @nimctl for the review. :nerd_face: 

I updated it to include the `description` in all the security scheme classes and added tests for them all.

Thanks for your contribution! :rocket: :cake:

**Pull request merged**